src/plugins/data/common/search/search_source/fetch/types.ts

```diff
@@ -17,7 +19,12 @@ import { GetConfigFn } from '../../../types';
  * where `ISearchRequestParams` is used externally instead.
  * FIXME: replace with estypes.SearchRequest?
  */
-export type SearchRequest = Record<string, any>;
+export interface SearchRequest {
```

Comment:
Here's an idea: given the comment above, I wonder if it would be possible to make it a little bit nicer the type here to let the consumer extends the dynamic typings at will:

```ts
export type SearchRequest<T extends Record<string, any> = Record<string, any>> = {
  index?: DataView | string;
  query?: Array<Query | AggregateQuery>;
  filters?: Filter[] | (() => Filter[]);
} & Omit<T, 'index' | 'query' | 'filters'>;
```

The type definition above gives an equivalent IntelliSense UX than the one in the PR, but the consumer can also decide to be more specific with the generic type:

```ts
const request: SearchRequest<estypes.SearchRequest> = { ... };
```